hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -567,32 +587,27 @@ private static Schema visitInternalPrimitiveToBuildAvroPrimitiveType(Type.Primit
         return Schema.createFixed(name, null, null, fixed.getFixedSize());
       }
 
-      case DECIMAL: {
-        Types.DecimalType decimal = (Types.DecimalType) primitive;
+      case DECIMAL:
+      case DECIMAL_FIXED: {
+        Types.DecimalTypeFixed decimal = (Types.DecimalTypeFixed) primitive;
         // NOTE: All schemas corresponding to Avro's type [[FIXED]] are generated
         //       with the "fixed" name to stay compatible w/ [[SchemaConverters]]
         String name = recordName + AVRO_NAME_DELIMITER + "fixed";
         Schema fixedSchema = Schema.createFixed(name,
-            null, null, computeMinBytesForPrecision(decimal.precision()));
+            null, null, decimal.getFixedSize());
```

Comment:
Is `decimal.getFixedSize()` the same as `computeMinBytesForPrecision`?

Reply:
Yes